### PR TITLE
Update CAPZ roadmap

### PR DESCRIPTION
**Pull request by dtzar**
Signed-off-by: David Tesar <david.tesar@microsoft.com>

*What type of PR is this?**

/kind documentation

**What this PR does / why we need it**:
Update the roadmap to current

### CODE REVIEW

File: docs/book/src/roadmap.md [hunk 1]
```diff
@@ -1,21 +1,27 @@
...
-    - Includes: [ManagedClusters E2E Tests](https://github.com/kubernetes-sigs/cluster-api-provider-azure/issues/2873), [ClusterClass support](https://github.com/kubernetes-sigs/cluster-api-provider-azure/issues/2684), [Enabling all AKS features](https://github.com/kubernetes-sigs/cluster-api-provider-azure/issues/2625), [Evolution of standardized CAPI ManagedCluster for CAPZ](https://github.com/kubernetes-sigs/cluster-api/blob/main/docs/proposals/20220725-managed-kubernetes.md)
--  Latest features for self-managed - These larger or essential features enable a better experience for provisioning self-managed clusters on Azure.
-    - Includes: MachinePools support, Azure CNI v2 support
+- ManagedClusters - Provisioning new AKS (ManagedClusters) clusters at scale is a common use case we want to enable an excellent experience with stability and all of the same features available in AKS. Includes:
+  - [ClusterClass support](https://github.com/kubernetes-sigs/cluster-api-provider-azure/issues/2684)
+  - [Enabling all AKS features](https://github.com/kubernetes-sigs/cluster-api-provider-azure/issues/3446)
```
Comment (CecileRobertMichon) [inline docs/book/src/roadmap.md:18]: Wonder if enabling preview features (https://github.com/kubernetes-sigs/cluster-api-provider-azure/issues/2625) should get its own bullet point?

**Review by mboersma — COMMENTED** [review]
/lgtm
/release-note-none

**Review by nojnhuh — COMMENTED** [review]
/lgtm
/approve

**Comment by nojnhuh** [comment]
@dtzar Should this be backported now so we get in on the website before v1.12?

**Pull request merged**